bigchaindb/util.py

```diff
@@ -193,16 +316,94 @@ def verify_signature(signed_transaction):
         bool: True if the signature is correct, False otherwise.
     """
 
-    data = signed_transaction.copy()
+    for fulfillment in signed_transaction['transaction']['fulfillments']:
+        fulfillment_message = get_fulfillment_message(signed_transaction, fulfillment)
+        # verify the fulfillment (for now lets assume there is only one owner)
+        try:
+            parsed_fulfillment = Fulfillment.from_uri(fulfillment['fulfillment'])
+        except Exception:
+            return False
```

Comment:
Would it be possible to use explicit exceptions?

see the section on exceptions in https://www.python.org/dev/peps/pep-0008/#programming-recommendations